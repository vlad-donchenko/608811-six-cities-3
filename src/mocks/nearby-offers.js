const nearbyOffers = [
  {
    bedrooms: 5,
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century`,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Baby seat`],
    host: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 50,
      isPro: true,
      name: `Angelina`
    },
    id: 50,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: false,
    isPremium: true,
    location: {
      latitude: 52.398298,
      longitude: 4.848328,
      zoom: 8
    },
    maxAdults: 3,
    previewImage: `apartment-01.jpg`,
    price: 424,
    rating: 4,
    title: `Beautiful & luxurious apartment at great location - Nearby`,
    type: `Private room`
  },
  {
    bedrooms: 5,
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century`,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Baby seat`],
    host: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 51,
      isPro: true,
      name: `Angelina`
    },
    id: 51,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.397288,
      longitude: 4.852012,
      zoom: 8
    },
    maxAdults: 3,
    previewImage: `apartment-01.jpg`,
    price: 124,
    rating: 2,
    title: `Beautiful & luxurious apartment at great location - Nearby`,
    type: `Private room`
  },
  {
    bedrooms: 5,
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century`,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Baby seat`],
    host: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 52,
      isPro: true,
      name: `Angelina`
    },
    id: 52,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: true,
    isPremium: true,
    location: {
      latitude: 52.395049,
      longitude: 4.854962,
      zoom: 8
    },
    maxAdults: 3,
    previewImage: `apartment-02.jpg`,
    price: 924,
    rating: 2,
    title: `Beautiful & luxurious apartment at great location - Nearby`,
    type: `Private room`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 48.866411,
        longitude: 2.331896,
        zoom: 15
      },
      name: `Paris`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Wi-Fi`, `Kitchen`, `Towels`, `Baby seat`],
    host: {
      avatarUrl: `avatar-max.jpg`,
      id: 53,
      isPro: true,
      name: `Max`
    },
    id: 53,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 48.869508,
      longitude: 2.327385,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `apartment-02.jpg`,
    price: 1100,
    rating: 1,
    title: `Sofitel Paris Le Faubourg - Nearby`,
    type: `Apartment`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 48.866411,
        longitude: 2.331896,
        zoom: 15
      },
      name: `Paris`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Wi-Fi`, `Kitchen`, `Towels`, `Baby seat`],
    host: {
      avatarUrl: `avatar-max.jpg`,
      id: 54,
      isPro: true,
      name: `Max`
    },
    id: 54,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: true,
    isPremium: true,
    location: {
      latitude: 48.870955,
      longitude: 2.330679,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `apartment-02.jpg`,
    price: 100,
    rating: 5,
    title: `Sofitel Paris Le Faubourg - Nearby`,
    type: `Apartment`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 48.866411,
        longitude: 2.331896,
        zoom: 15
      },
      name: `Paris`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Wi-Fi`, `Kitchen`, `Towels`, `Baby seat`],
    host: {
      avatarUrl: `avatar-max.jpg`,
      id: 55,
      isPro: true,
      name: `Max`
    },
    id: 55,
    images: [`room.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `apartment-01.jpg`, `apartment-01.jpg`],
    isFavorite: false,
    isPremium: true,
    location: {
      latitude: 48.869918,
      longitude: 2.327986,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `apartment-03.jpg`,
    price: 190,
    rating: 3,
    title: `Sofitel Paris Le Faubourg - Nearby`,
    type: `Apartment`
  }
];

export default nearbyOffers;
