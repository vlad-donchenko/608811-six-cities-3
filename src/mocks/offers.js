import {getRandomInteger} from "../utils/index";

const INDICATOR = 0.5;

const titles = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
  `Wood and stone place`
];

const images = [
  `room.jpg`,
  `apartment-02.jpg`,
  `apartment-03.jpg`,
  `apartment-01.jpg`,
  `apartment-01.jpg`,
];

const goods = [
  `Wi-Fi`,
  `Heating`,
  `Kitchen`,
  `Fridge`,
  `Washing machine`,
  `Coffee machine`,
  `Dishwasher`,
  `Towels`,
  `Baby seat`,
  `Cabel TV`
];

const names = [
  `Angelina`,
  `Max`,
  `Sam`,
  `Nick`,
  `Jane`
];

const avatars = [
  `avatar-angelina.jpg`,
  `avatar-max.jpg`
];

const descriptions = [
  `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century`,
  `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful`,
  `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century`
];

const latitudes = [
  52.3909553943508,
  52.369553943508,
  52.3909553943508,
  52.3809553943508,
  52.3709553943508
];

const longitude = [
  4.85309666406198,
  4.85309666406198,
  4.929309666406198,
  4.939309666406198,
  4.979309666406198
];

const getGoods = () => goods.filter(() => Math.random() > INDICATOR);

const getDescription = () => {
  return descriptions.filter(() => {
    return Math.random() > INDICATOR;
  }).join(`. `);
};

const generateOffers = () => {
  return titles.map((title, id) => {
    return {
      bedrooms: getRandomInteger(1, 7),
      city: {
        location: {
          latitude: 52.38333,
          longitude: 4.9,
          zoom: 10,
        },
        name: `Amsterdam`
      },
      description: getDescription(),
      goods: getGoods(),
      host: {
        avatarUrl: avatars[getRandomInteger(0, avatars.length - 1)],
        id,
        isPro: Math.random() > INDICATOR,
        name: names[id]
      },
      id,
      images,
      isFavorite: Math.random() > INDICATOR,
      isPremium: Math.random() > INDICATOR,
      location: {
        latitude: latitudes[id],
        longitude: longitude[id],
        zoom: 8
      },
      maxAdults: getRandomInteger(1, 7),
      previewImage: images[getRandomInteger(0, images.length - 1)],
      price: getRandomInteger(100, 500),
      rating: getRandomInteger(0, 5),
      title,
      type: Math.random() > INDICATOR ? `Private room` : `Apartment`,
    };
  });
};

const offers = generateOffers();

export {offers, names};
