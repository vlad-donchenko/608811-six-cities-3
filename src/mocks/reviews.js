const reviews = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 1,
      isPro: false,
      name: `Max`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-06-08T14:13:56.569Z`,
    id: 2,
    rating: 5,
    user: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 2,
      isPro: true,
      name: `Angelina`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-09-08T14:13:56.569Z`,
    id: 3,
    rating: 1,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 3,
      isPro: false,
      name: `Max`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-10-08T14:13:56.569Z`,
    id: 4,
    rating: 5,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 4,
      isPro: false,
      name: `Angelina`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-10-08T14:13:56.569Z`,
    id: 5,
    rating: 2,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 3,
      isPro: true,
      name: `Max`
    }
  }
];

export {reviews};
