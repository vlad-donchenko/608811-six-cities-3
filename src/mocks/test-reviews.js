const reviews = [
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-08-08T14:13:56.569Z`,
    id: 1,
    rating: 5,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 1,
      isPro: true,
      name: `Max`
    }
  },
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-06-03T14:13:56.569Z`,
    id: 2,
    rating: 4,
    user: {
      avatarUrl: `avatar-angelina.jpg`,
      id: 2,
      isPro: false,
      name: `Angelina`
    }
  },
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-06-01T14:13:56.569Z`,
    id: 3,
    rating: 2,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 3,
      isPro: false,
      name: `Max`
    }
  },
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-11-08T14:13:56.569Z`,
    id: 4,
    rating: 2,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 4,
      isPro: false,
      name: `Angelina`
    }
  },
  {
    comment: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    date: `2019-01-08T14:13:56.569Z`,
    id: 5,
    rating: 4,
    user: {
      avatarUrl: `avatar-max.jpg`,
      id: 3,
      isPro: false,
      name: `Max`
    }
  }
];

export {reviews};
