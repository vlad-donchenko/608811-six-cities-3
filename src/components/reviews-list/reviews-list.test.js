import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list";
import {offers} from "../../mocks/test-offer";

it(`Should render <ReviewsItem/>`, () => {
  const tree = renderer.create(<ReviewsList reviews={offers[0].reviews}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
