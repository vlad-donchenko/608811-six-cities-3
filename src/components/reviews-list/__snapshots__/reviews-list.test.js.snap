// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <ReviewsItem/> 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      5
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Angelina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              2
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-10-08"
        >
          November 8 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              5
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-07-08"
        >
          August 8 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Angelina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              4
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-03"
        >
          June 3 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              2
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-01"
        >
          June 1 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              4
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-00-08"
        >
          January 8 2019
        </time>
      </div>
    </li>
  </ul>
</section>
`;
